src/v/cluster/topic_table.cc

```diff
@@ -1364,18 +1206,6 @@ ss::future<> topic_table::apply_snapshot(
 
     // 2. re-calculate derived state
 
-    _topics_hierarchy.clear();
-    for (const auto& [ns_tp, md_item] : _topics) {
-        if (!md_item.metadata.is_topic_replicable()) {
-            // NOTE: here we assume that source and derivative topics are in the
-            // same namespace.
-            model::topic_namespace_view src_ns_tp(
-              ns_tp.ns, md_item.metadata.get_source_topic());
-            _topics_hierarchy[src_ns_tp].insert(ns_tp);
-        }
-        co_await ss::coroutine::maybe_yield();
-    }
-
     _partition_count = 0;
     for (const auto& [ns_tp, md_item] : _topics) {
         if (md_item.metadata.is_topic_replicable()) {
```

Comment:
To check my understanding:

Can this whole loop become a `std::accumulate` and doesn't need to check this now right (`is_topic_replicable` is always true right?)

Does that mean we can remove `is_topic_replicable` from `topic_metadata`? Or should we do that in another PR? 😄

Reply:
good point, we should remove it

Reply:
i have those changes stashed i thought i already committed them